### ISSUE TITLE
Nix CI of MathComp should run the test suites of Mczify and Algebra Tactics

### ISSUE THREAD

**Issue by pi8027** [math-comp/math-comp#1219]
Since these two packages provide proof-automation functionalities, checking whether they compile is insufficient. At the very least, we should run the test suites (`make test-suite`). In fact, recent removal of the `nat_rec_scope` (#1162) broke the test suite of Mczify. (math-comp/mczify#55)

**Comment by CohenCyril** [comment]
Agreed! I do not know how to implement that yet in the coq-nix-toolbox... Maybe the test suite should be added as an extra derivation? This is a general coq-nix-toolbox question though...

**Comment by pi8027** [comment]
Hi @proux01 @CohenCyril. IIRC, the plan was to fix this properly during Rocq’n’share. I'm not in Paris, but I'm available for video calls.

**Comment by Zimmi48** [comment]
> > but I wonder if packages compiled with a `doCheck` are different than the ones without...
> 
> ~[This](https://nixos.org/manual/nixpkgs/stable/#tester-testEqualDerivation) seems to answer my question negatively, ... which is good!~ Actually it seems it just means that `doCheck = true` should not alter the result, but not that nix enforces it 😕

I am quite convinced that derivations with `doCheck = true` vs `doCheck = false` will be different, otherwise they would behave the same. Therefore (given that we don't have a content-addressed Nix store by default), there won't be any caching of the results between the two (if that's what you hoped for). If you want caching, then you have to separate the test phase into a different derivation, which is possible as well.
